asistente_ladm_col/gui/controlled_measurement_dialog.py

```diff
@@ -114,31 +129,151 @@ def accept_dialog(self):
+                layer.fields().indexOf("y_stdev"): y_stdev
+            })
+            new_feature = QgsVectorLayerUtils.createFeature(layer, geom, fields_values)
             new_features.append(new_feature)
 
         layer.dataProvider().addFeatures(new_features)
+        features = groups.getFeatures("\"{}\" IS NULL".format("belongs_to_group"))
+        layer.dataProvider().addFeatures(features)
+        layer.commitChanges()
         QgsProject.instance().addMapLayer(layer)
 
         self.qgis_utils.message_emitted.emit(
             QCoreApplication.translate("ControlledMeasurementDialog",
                                        "A new average point layer has been added to the map!"),
             Qgis.Info)
 
+    def run_group_points_model(self, input_layer, tolerance, definition_field):
+        # Run model
+        model = QgsApplication.processingRegistry().algorithmById("model:Group_Points")
+        if model:
+            params = {
+                '1_Inputpoints': input_layer.source(),
+                '2_Typedefinition': definition_field,
+                '3_Tolerance': tolerance,
+                'native:multiparttosingleparts_2:output': 'memory:',
+                'native:mergevectorlayers_1:output': 'memory:'
+            }
+            res = processing.run("model:Group_Points", params)
+            msg = "Model Group_Points and execute OK!"
+            return res, msg
+        else:
+            res = None
+            msg = "Model Group_Points was not found and cannot be opened!"
+            return res, msg
+
+    def time_validate(self, input_features, time_tolerance, time_field):
+        groups_num = input_features.uniqueValues(input_features.fields().indexFromName("belongs_to_group"))
+        idx = input_features.fields().indexFromName(time_field)  # Cambiar por el campo de tiempo cuando este se incluya
+        new_layer = self.copy_attribs(input_features)
+        time_layer = self.copy_attribs(input_features)
+        time2_layer = self.copy_attribs(input_features)
+        for i in groups_num:
+            if i is None:
+                #new_layer.startEditing()
+                features = input_features.getFeatures("\"{}\" IS NULL".format("belongs_to_group"))
+                new_layer.dataProvider().addFeatures(features)
+                [new_layer.changeAttributeValue(feat.id(), feat.fields().indexOf("trusty"), "False") for feat in
+                 new_layer.getFeatures()]
+                new_layer.commitChanges()
+            else:
+                features, not_features = self.time_filter(input_features, input_features.getFeatures("\"{}\"={}".format("belongs_to_group", i)), idx, time_tolerance)
+                features = [f for f in features]
+                not_features = [f for f in not_features]
+                fg = len(features)
+                if fg > 1:
+                    time_layer.startEditing()
+                    time_layer.dataProvider().addFeatures(features)
```

Comment:
If you open an edit session, you should avoid using dataProvider() to perform changes over that layer. 
Choose one (either edit session or dataProvider) and stick to it.

Reply:
solved in the commit: https://github.com/AgenciaImplementacion/Asistente-LADM_COL/commit/41a6511b624039fa0bb91702dbef2206a0e1bc83